uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader.php

```diff
@@ -152,6 +153,34 @@ private function deleteLocationTypes() {
       'name' => ['IN' => $locationsToDelete],
       'api.LocationType.delete' => ['id' => "\$value.id"]
     ]);
+
+  }
+
+  /**
+   * Creates location types required by CiviHR and makes them reserved
+   */
+  private function createRequiredLocationTypes() {
+    $locationTypesToCreate = [
+      'Personal',
```

Comment:
Since the Personal type is created on install, What about existing sites, is an upgrader doing this somewhere?

Reply:
Already answered this in chat, but just for posterity the "Personal" type is actually created from [buildkit](https://github.com/civicrm/civicrm-buildkit/blob/a4735eb44f89ef28ce3d2153a6084026c7804f77/app/config/hr17/install.sh#L27). Because some sites could have potentially deleted it we agreed that I would add an upgrader function to ensure the "Personal" location type exists and is in the right state.